packages/maskbook/src/plugins/PoolTogether/utils.ts

```diff
@@ -0,0 +1,50 @@
+import { ChainId } from '@masknet/web3-shared'
+import { ONE_SECOND, ONE_WEEK_SECONDS } from './constants'
+import type { Pool } from './types'
+
+export const calculateOdds = (usersTicketBalance: number, totalSupply: number, numberOfWinners: number) => {
+    if (usersTicketBalance === 0 || Number.isNaN(usersTicketBalance)) {
+        return undefined
+    }
+
+    // Calculate odds of winning at least 1 of the possible scenarios.
+    // 1/N, 2/N ... N-1/N, N/N
+    // Then we always display "1 in ____" so 1 / X.
+    const odds = 1 / (1 - Math.pow(totalSupply / (totalSupply + usersTicketBalance), numberOfWinners))
+
+    return odds.toLocaleString(undefined, {
+        maximumFractionDigits: odds >= 10 ? 0 : 8,
+    })
+}
+
+export const calculateNextPrize = (pool: Pool) => {
+    const rawPrize =
+        Number.parseInt(pool.prize.weeklyTotalValueUsd, 10) /
+        (ONE_WEEK_SECONDS / Number.parseInt(pool.config.prizePeriodSeconds, 10))
+
+    if (!Number.isNaN(rawPrize)) {
+        return `\$${rawPrize.toLocaleString(undefined, { maximumFractionDigits: 0 })}`
+    } else {
+        const pirzeAmount = Number.parseFloat(pool.prize.amount)
+        return `${pirzeAmount.toLocaleString(undefined, {
+            maximumFractionDigits: pirzeAmount >= 10 ? 0 : 2,
+        })} ${pool.tokens.underlyingToken.symbol}`
```

Comment:
```suggestion
        const pirzeAmount = Number.parseFloat(pool.prize.amount).toLocaleString(undefined, {
            maximumFractionDigits: pirzeAmount >= 10 ? 0 : 2,
        })
        return `${pirzeAmount} ${pool.tokens.underlyingToken.symbol}`
```

Reply:
Can't do that, need `prizeAmount` for formatting.